assets/js/components/dashboard-sharing/ModuleRecoveryAlert/index.js

```diff
@@ -241,29 +329,31 @@ export default function ModuleRecoveryAlert() {
 				{ Object.keys( recoveryErrors ).length > 0 && (
 					<Errors recoveryErrors={ recoveryErrors } />
 				) }
+				<ActionsCTALinkDismiss
+					id={ id }
+					ctaLabel={ __( 'Recover', 'google-site-kit' ) }
+					onCTAClick={ handleRecoverModules }
+					isSaving={ recoveringModules }
+					dismissLabel={ __( 'Remind me later', 'google-site-kit' ) }
+					dismissOnCTAClick={ shouldDismissOnCTAClick }
+					dismissExpires={ DAY_IN_SECONDS }
+					ctaDismissOptions={ { skipHidingFromQueue: false } }
+					ctaDisabled={ disableCTA }
+				/>
 			</Fragment>
 		);
-		onCTAClick = handleRecoverModules;
 	}
 
 	return (
-		<BannerNotification
-			id="module-recovery-alert"
-			title={ __(
-				'Dashboard data for some services has been interrupted',
-				'google-site-kit'
-			) }
-			onCTAClick={ onCTAClick }
-			ctaLabel={ onCTAClick ? __( 'Recover', 'google-site-kit' ) : null }
-			ctaLink={ onCTAClick ? '#' : null }
-			description={ description }
-			learnMoreURL={ documentationURL }
-			learnMoreLabel={ __( 'Learn more', 'google-site-kit' ) }
-			isDismissible={ isDismissible }
-			dismiss={ __( 'Remind me later', 'google-site-kit' ) }
-			dismissExpires={ DAY_IN_SECONDS }
-		>
-			{ children }
-		</BannerNotification>
+		<Notification className="googlesitekit-publisher-win">
+			<SimpleNotification
+				title={ __(
+					'Dashboard data for some services has been interrupted',
+					'google-site-kit'
+				) }
+				description={ description }
+				actions={ actions }
+			/>
+		</Notification>
```

Comment:
I've refactored the component significantly, it's longer in terms of lines because props in components such as `ActionsCTALinkDismiss` and `Dismiss` those that were passed to the `BannerNotification` component. The implementation here is clear and now reuses key shared notification components.

Reply:
Thanks @benbowler, I spent a bit looking into what a more substantial refactor might look like for this component as it's quite complex but I think it's probably best saved for its own issue.

I'd love to see a few components and/or hooks extracted here for the various `description` and `actions` cases above. We should avoid having large nested if statements as it's quite hard to read and understand. Similarly, the learn more link is the same in all cases where it's present.

Feel free to address any low-hanging fruit you see if time allows, but otherwise, let's open a follow up issue to (probably rewrite) this component.